### PR TITLE
Update SQL Server schema guide documentation

### PR DESCRIPTION
**Pull request by rbans96**
## Description
Updates the SQL Server schema migration guide documentation

## Related issues
Addresses #75040

## Testing
N/A

### CODE REVIEW

File: docs/SchemaMigrationGuide.md [hunk 1]
```diff
@@ -1,53 +1,54 @@
...
 
 - ### Prerequisites
 
     - Application should have admin access to the database.
 
-- ### When might you choose this option
-    -  There is no specific DBA role - In this case, automatic schema upgrade is a good choice as it just requires the application with admin access to the database.
+- ### When might you choose this method?
 
     - Non-Production scenarios - The auto upgrade is a good option for dev/tests environments. This will automatically apply any pending migrations when the application launches. Hence, the testing of schema upgrade becomes part of the test cycle for the application.
```
Comment (brendankowitz) [inline docs/SchemaMigrationGuide.md:27]: `dev/test`

File: docs/SchemaMigrationGuide.md [hunk 2]
```diff
@@ -1,53 +1,54 @@
 # Schema Migration Guide
-This document describes how to perform Sql schema migration on the application server.
+As Microsoft continues to support the newer releases of FHIR, migrations for the FHIR Server for Azure SQL Server edition will be required.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:2]: It might make sense to move the description of what a migration is a bit further up in the documentation, before you start using the word "migration".

Maybe something like this:

`As Microsoft continues to develop the FHIR Server for Azure SQL Server and support new releases of FHIR, updates will need to be made to the T-SQL script that alters the SQL database. 
This document describes how to perform an update to the T-SQL script on the FHIR Server for Azure application server. This is referred to as a SQL Server schema migration.`

File: docs/SchemaMigrationGuide.md [hunk 3]
```diff
@@ -1,53 +1,54 @@
...
 
-## 1. Automatic Schema Upgrade
-Here, the schema migration is performed automatically on the server startup.
-The scripts that upgrade schema are part of any upgrade package for the application and run automatically when the application launches.
+## 1. Automatic schema upgrade
+When using the automatic schema upgrade method, the SQL Server schema migration is performed automatically on the FHIR Server for Azure upgrade.
+The scripts which upgrade the SQL Server schema are part of any upgrade package for the web application and run automatically when the web application launches.
 
-In case of any errors during upgrade, the application fails to start and the upgrade script transaction is not committed.
+In case of any errors during upgrade, the web application fails to start and the upgrade script transaction is not committed.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:19]: Nit: maybe rephrase this slightly to say that the application will not start or that it will fail safely?

File: docs/SchemaMigrationGuide.md (hunk 1 above)
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:27]: Nice, I really like how this section is worded and I think it will be helpful

File: docs/SchemaMigrationGuide.md [hunk 4]
```diff
@@ -1,53 +1,54 @@
...
 
-2. Schema upgrade via schema migration tool
+2. Schema upgrade via SQL Server schema migration tool
 
-## 1. Automatic Schema Upgrade
-Here, the schema migration is performed automatically on the server startup.
-The scripts that upgrade schema are part of any upgrade package for the application and run automatically when the application launches.
+## 1. Automatic schema upgrade
+When using the automatic schema upgrade method, the SQL Server schema migration is performed automatically on the FHIR Server for Azure upgrade.
+The scripts which upgrade the SQL Server schema are part of any upgrade package for the web application and run automatically when the web application launches.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:17]: Minor suggestion: `The scripts which upgrade the SQL Server schema could be part of any upgrade package`

File: docs/SchemaMigrationGuide.md [hunk 5]
```diff
@@ -1,53 +1,54 @@
...
 
-In case of any errors during upgrade, the application fails to start and the upgrade script transaction is not committed.
+In case of any errors during upgrade, the web application fails to start and the upgrade script transaction is not committed.
 
 - ### Prerequisites
 
     - Application should have admin access to the database.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:23]: Should this also be added to the schema tool prerequisites?

File: docs/SchemaMigrationGuide.md [hunk 6]
```diff
@@ -1,53 +1,54 @@
...
 
     - Non-Production scenarios - The auto upgrade is a good option for dev/tests environments. This will automatically apply any pending migrations when the application launches. Hence, the testing of schema upgrade becomes part of the test cycle for the application.
 
-## 2. Upgrade via Schema migration tool
-Schema migration tool is a command-line utility that admins can run to perform schema migrations on demand.
+## 2. Upgrade via SQL Server schema migration tool
+The SQL Server schema migration tool is a command-line utility which admins can run to perform SQL Server schema migrations on demand.
 
-Schema upgrade scripts would still be part of any upgrade package for the application but the scripts will not run automatically on the server startup. In order to perform the schema upgrade, the schema admin would need to run the schema migration tool.
+The SQL Server schema upgrade scripts would still be part of any upgrade package for the web application but the scripts will not run automatically on the FHIR server for Azure upgrade. To perform the SQL Server schema upgrade, the schema admin would need to run the SQL Server schema migration tool.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:32]: Sorry for all these nit picks! - but I think it would be good to capitalize the "S" for FHIR Server for Azure

File: docs/SchemaMigrationGuide.md [hunk 7]
```diff
@@ -1,53 +1,54 @@
...
 
  - ### Prerequisites
 
-    - The schema migration tool is available.
+    - The SQL Server schema migration tool is available.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:36]: What do you mean by available, exactly?

File: docs/SchemaMigrationGuide.md [hunk 8]
```diff
@@ -1,53 +1,54 @@
...
+ - ### When might you choose this method?
     - Production environment
  
-        It is highly recommended to use schema migration tool for the production environment to avoid downtime. Automatic schema migrations can take a long time, often in proportion to the size of the database. For this reason, performing schema migrations on startup may not be a good option.
+        It is highly recommended to use SQL Server schema migration tool for the production environment to avoid downtime. Automatic SQL Server schema migrations can take a long time, often in proportion to the size of the database. For this reason, performing SQL Server schema migrations on startup may not be a good option.
  
     - Role-based security
  
-        Only Database/Schema Admin can run this tool to upgrade the schema. It allows you to setup proper roles and permissions for the application e.g. the application is assigned the role as only Database Reader/Writer.
+        Only Database/Schema Admin can run this tool to upgrade the SQL Server schema. It allows to setup proper roles and permissions for the web application. For example - the application is assigned the role as only Database Reader/Writer.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:45]: Nit: `It allows for proper setup of roles and permissions`

Comment (rotodd) [reply docs/SchemaMigrationGuide.md:45]: Nit: `as Database Reader/Writer only.`

File: docs/SchemaMigrationGuide.md [hunk 9]
```diff
@@ -1,53 +1,54 @@
...
 
-- ### Schema migration tool
+- ### SQL Server schema migration tool
 
     The detailed information about this tool is provided [here](SchemaMigrationTool.md).
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:49]: +1 it's great that we have this

File: docs/SchemaMigrationGuide.md [hunk 10]
```diff
@@ -66,24 +67,19 @@ There is a configurable property in the launchSettings.json which can be set to
...
+ A1. The FHIR Server for Azure app service is upgraded first. Because firstly it contains the scripts to migrate the SQL Server schema and secondly the code is considered to work for a number of SQL Server schema versions and will specify the min and max supported versions.
 
- Q2. What should we do in case of automatic schema upgrade fails?
+ Q2. What should we do in case of automatic SQL Server schema upgrade fails?
 
- Ans. If automatic schema upgrade fails and application is unable to start then you can toggle the schema auto upgrade flag as mentioned under the section (How to toggle between the options) so that the upgrade scripts would not run and application would start on the current version.
- Also, you can reach out to the concerned team with the error description to get further help on the issue.
+ A2. If automatic SQL Server schema upgrade fails and web application is unable to start then you can toggle the SQL Server schema auto upgrade flag as mentioned under the section (How to toggle between the options) so that the upgrade scripts would not run and application would start on the current version.
```
Comment (rotodd) [inline docs/SchemaMigrationGuide.md:85]: Very minor but: `If an automatic` and `the web application`

Comment (rotodd) [reply docs/SchemaMigrationGuide.md:85]: I believe you might also be able to put a hyperlink to the "How to toggle between the options" section of the document if you are feeling like being a documentation nerd like me 😎

File: docs/SchemaMigrationGuide.md (hunk 5 above)
Comment (rbans96) [reply docs/SchemaMigrationGuide.md:23]: Here is an assumption that the tool would only be run by Schema admin, So don't think so if it should be mentioned under tool prerequisites.

File: docs/SchemaMigrationGuide.md (hunk 7 above)
Comment (rbans96) [reply docs/SchemaMigrationGuide.md]: Thanks for checking. I have reworded it to say installed and available to use.
